File: common/src/main/java/juuxel/adorn/client/gui/screen/AdornMenuScreen.java
package juuxel.adorn.client.gui.screen;

import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.player.Inventory;
import net.minecraft.world.inventory.AbstractContainerMenu;

public abstract class AdornMenuScreen<M extends AbstractContainerMenu> extends AbstractContainerScreen<M> {
    public AdornMenuScreen(M menu, Inventory playerInventory, Component title) {
        super(menu, playerInventory, title);
    }

    public AdornMenuScreen(M menu, Inventory playerInventory, Component title, int imageWidth, int imageHeight) {
        super(menu, playerInventory, title, imageWidth, imageHeight);
    }

    public int getPanelX() {
        return leftPos;
    }

    public int getPanelY() {
        return topPos;
    }
}

File: common/src/main/kotlin/juuxel/adorn/client/gui/screen/AdornMenuScreen.kt
package juuxel.adorn.client.gui.screen

import net.minecraft.client.gui.DrawContext
import net.minecraft.client.gui.screen.ingame.MenuScreen
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.menu.Menu
import net.minecraft.text.Text

abstract class AdornMenuScreen<M : Menu>(
    menu: M,
    playerInventory: PlayerInventory,
    title: Text
) : MenuScreen<M>(menu, playerInventory, title) {
    val panelX get() = x
    val panelY get() = y

    override fun render(context: DrawContext, mouseX: Int, mouseY: Int, delta: Float) {
        renderBackground(context, mouseX, mouseY, delta)
        super.render(context, mouseX, mouseY, delta)
        drawMouseoverTooltip(context, mouseX, mouseY)
    }
}
